### ISSUE TITLE
Not writing metadata to archive

### ISSUE THREAD

**Issue by shrublet** [jvlflame/FkuLibrary#3]
Not sure why I'm encountering this issue, but it seems that it doesn't write the ComicInfo.xml to the archive but rather the directory the book is in. While running, it also says it can't find the book (WARNING xyz not found on FAKKU/panda.chaika), but the ComicInfo.xml seems to be pulling the correct info. I tried to individually set the metadata of the books using -Url, but the same issue occurs. I tried with the Powershell 5.1 and 7.1 but both produce the same results. I've also tried testing on the same drive as FkuLibrary as well as my secondary drive but to no avail. I'm on Windows 11 if that matters, but I hope it isn't affecting this. Not sure what else I can try so help much appreciated! Thanks!

**Comment by jvlflame** [comment]
Awesome! I'm glad you were able to resolve it.

> There was also the issue of SetMetadataXML using $NewName as the DestinationPath flag for Compress-Archive, but if the archive did not contain any brackets, $NewName is never defined. 

That seems to be an oversight on my part.. good catch. I probably didn't catch it because the majority of manga that come from the torrents have brackets in them.

>  I can send the code if needed, but I'm not sure if this is an isolated fix/issue still or if my diagnosis is even correct tbh lol.

Your diagnosis does seem correct! I'll gladly accept a PR if you're up to it.

> Edit: Also realizing that, at least for me, Compress-Archive doesn't like .cbz extensions and only works with .zip.

The [Get-LocalArchives](https://github.com/jvlflame/FkuLibrary/blob/master/internal/Get-LocalArchives.ps1) function does currently detect multiple archive extension types including `.cbz`, but I'm surprised `Compress-Archive` doesn't work for you. I just tested on my system and it does work. You could probably add a workaround in `Set-FakkuMetadata` to update the extension to `.zip` before the `Compress-Archive` and then change it back after.

**Comment by shrublet** [comment]
Honestly I changed like two lines of code max so I'm don't it's worth a PR lol. That and there's issues I'm encountering that I can't fully confirm if they're purely personal or not since you said it's properly working for you.

 I added an else clause on line 19 of [SetMetadataXML ](https://github.com/jvlflame/FkuLibrary/blob/40eb297aa2913e8d172fd038065e2e366ada3a16/internal/Set-MetadataXML.ps1#L19)   `else { $NewName = $FilePath }` to account for bracketless names. To fix the file path issue, on line 91 and 107 of [Set-FakkuMetadata](https://github.com/jvlflame/FkuLibrary/blob/40eb297aa2913e8d172fd038065e2e366ada3a16/functions/Set-FakkuMetadata.ps1#L91), I just used `Set-MetadataXML -FilePath $File -XMLPath $XMLPath -Content $xml` instead of `Set-MetadataXML -FilePath **$File.Name** -XMLPath $XMLPath -Content $xml` since `Compress-Archive` needs a full path and `$File.Name` would only pass the name.

I've also encountered issues with [Get-FakkuUrl](https://github.com/jvlflame/FkuLibrary/blob/master/internal/Get-FakkuUrl.ps1) not properly sterilizing artist names in brackets. It seems in [this commit](https://github.com/jvlflame/FkuLibrary/commit/269eaf04fd8ef6d2cf609dc83c2afadc26fd7122#diff-7bdf9b9d5715c5efff94404441dee0e9f33a467f18153a1ff243313d7790cecc), on line 11,  you changed the regex matching which broke the deletion of text inside the brackets. I just reverted it to what you had previously and it fixed it and scraped proper links. As is, I was getting the artist name included in the links which doesn't abide by FAKKU naming convention (at least right now, not sure if it used to be). Hope this helps!
